src/test/java/org/gridsuite/modification/server/modifications/GeneratorCreationInBusBreakerTest.java

```diff
@@ -0,0 +1,135 @@
+                .ratedNominalPower(10.)
+                .activePowerSetpoint(400.)
+                .reactivePowerSetpoint(50.)
+                .voltageRegulationOn(true)
+                .voltageSetpoint(225.)
+                .stepUpTransformerReactance(60.0)
+                .transientReactance(61.0)
+                .minimumReactivePower(20.0)
+                .maximumReactivePower(25.0)
+                .droop(5f)
+                .participate(true)
+                .regulatingTerminalId("idGenerator1")
+                .regulatingTerminalType("GENERATOR")
+                .regulatingTerminalVlId("v1")
+                .qPercent(25.)
+                .reactiveCapabilityCurve(true)
+                .reactiveCapabilityCurvePoints(Arrays.asList(new ReactiveCapabilityCurveCreationInfos(2.0, 3.0, 3.1),
+                        new ReactiveCapabilityCurveCreationInfos(5.6, 9.8, 10.8)))
+                .connectionName("top")
+                .connectionDirection(ConnectablePosition.Direction.TOP)
+                .build();
+    }
+
+    @Override
+    protected ModificationInfos buildModificationUpdate() {
+        return GeneratorCreationInfos.builder()
+                .type(ModificationType.GENERATOR_CREATION)
+                .equipmentId("idGenerator2")
+                .equipmentName("nameGeneratorModified")
+                .voltageLevelId("v1")
+                .busOrBusbarSectionId("bus1")
+                .energySource(EnergySource.SOLAR)
+                .minActivePower(101.0)
+                .maxActivePower(601.0)
+                .ratedNominalPower(11.)
+                .activePowerSetpoint(401.)
+                .reactivePowerSetpoint(51.)
+                .voltageRegulationOn(true)
+                .voltageSetpoint(226.)
+                .stepUpTransformerReactance(61.0)
+                .transientReactance(62.0)
+                .minimumReactivePower(23.0)
+                .maximumReactivePower(26.0)
+                .droop(6f)
+                .participate(true)
+                .regulatingTerminalId("idGenerator1")
+                .regulatingTerminalType("GENERATOR")
+                .regulatingTerminalVlId("v1")
+                .qPercent(29.)
+                .reactiveCapabilityCurve(true)
+                .reactiveCapabilityCurvePoints(Arrays.asList(new ReactiveCapabilityCurveCreationInfos(1.0, 2.0, 2.1),
+                        new ReactiveCapabilityCurveCreationInfos(6.6, 8.8, 11.8)))
+                .connectionName("top")
+                .connectionDirection(ConnectablePosition.Direction.TOP)
+                .build();
+    }
+
+    @Override
+    protected MatcherGeneratorCreationInfos createMatcher(ModificationInfos modificationInfos) {
+        return MatcherGeneratorCreationInfos.createMatcherGeneratorCreationInfos((GeneratorCreationInfos) modificationInfos);
+    }
+
+    @Override
+    protected void assertNetworkAfterCreation() {
+        assertNotNull(getNetwork().getGenerator("idGenerator2"));
```

Comment:
you can also assert the VL update:
assertEquals(1, getNetwork().getVoltageLevel("v1").getGeneratorStream().filter(transformer -> transformer.getId().equals("idGenerator2")).count());

Reply:
Done